### PR TITLE
#GH-10 Using bot account to send standup report and notification instead of normal user

### PR DESCRIPTION
**Pull request by jatinjtg**
### Summary
Bot account is used to send the notification and standup report instead of normal user 

### Checklist
- [x] Added or updated test cases
- [x] `make test` Ran test cases and ensured they are passing
- [x] `make check-style` Ran style check and ensured both webapp and server components comply the style guides

### CODE REVIEW

File: server/config/main.go [hunk 1]
```diff
@@ -37,7 +36,8 @@ const (
 	// the date changed between 23:59 and 00:00:xx2.
 	RunnerInterval = 25 * time.Second
 
-	OverrideUsername = "Raven"
+	BotUsername = "raven"
```
Comment (harshilsharma63) [inline server/config/main.go]: What appears as the name besides the avatar depends on individual's setting (`Teamate Name Display`), can we do something to make sure the name always appears as `Raven` and not as `raven`?

File: server/plugin.go [hunk 2]
```diff
@@ -24,13 +25,17 @@ var SentryDSN string
...
-
+	BotUserID string
 	handler http.Handler
 	running bool
 }
 
 func (p *Plugin) OnActivate() error {
 	config.Mattermost = p.API
+	
+	if err := p.setupBot(); err !=nil {
```
Comment (harshilsharma63) [inline server/plugin.go]: Can we call this function from `OnConfigurationChange()` instead? This will remove the need of `Plugin.BotUserID` variable.

**Comment by harshilsharma63** [comment]
@jatinjtg I have one more question besides the review comments. What will happen with the existing posts created by standup raven? How would they look?

File: server/config/main.go (hunk 1 above)
Comment (jatinjtg) [reply server/config/main.go]: I have investigated on this but didn't find any way to do this as we can't use upper case in username.

**Comment by jatinjtg** [comment]
The drop in the test case coverage is because of the new test file plugin_test .

**Comment by harshilsharma63** [comment]
> The drop in the test case coverage is because of the new test file plugin_test .

Okay, got it.

**Pull request merged**